File: abc/abc168/abc168_c.fs
// https://atcoder.jp/contests/abc168/submissions/13389077

open System

let [| A; B; H; M |] = stdin.ReadLine().Split() |> Array.map float

let hour = 30. * H + 0.5 * M
let minute = 6. * M
let angle = abs (hour - minute)
let radian = angle * Math.PI / 180.

A * A + B * B - 2. * A * B * Math.Cos radian
|> Math.Sqrt
|> stdout.WriteLine

File: abc/abc168/abc168_c.cs
// https://atcoder.jp/contests/abc168/submissions/13394129

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine().Split().Select(double.Parse).ToArray();
            var A = input[0];
            var B = input[1];
            var H = input[2];
            var M = input[3];

            var hour = 30 * H + 0.5 * M;
            var minute = 6 * M;
            var angle = Math.Abs(hour - minute);
            var radian = angle * Math.PI / 180;

            Console.WriteLine(Math.Sqrt(A * A + B * B - 2 * A * B * Math.Cos(radian)));
        }
    }
}
